docs/TrafficSplitting.md

```diff
@@ -0,0 +1,155 @@
+- The order of the flags specified in the command are **not** taken into account.
+- The precedence of flag(s) as they are evaluated by `kn`:
+
+1. `--untag`: First, all the referenced tags by this flag are removed from the traffic block
+2. `--tag`: Second, all the revisions are tagged as specified in the traffic block
+3. `--traffic`: Third, all the referenced revisions (or tags) are assigned given portion of traffic split
+
+*Note*:
+
+- `--untag` flag's action takes first precedence. If `--untag` and `--tag` flags specified together for revisions, all the mentioned tags will be unassigned from revisions first followed by tagging them.
+- If one traffic block target has Revision name which matches Tag of another target, tag is looked up first for assigning traffic.
+  For example, `--traffic targetRef:Percent` will assign mentioned `Percent` to target where tag is `targetRef`.
+  Tags are unique in traffic block, Revision names aren't, use tag reference in such scenarios.
+- If a tag is assigned to a Revision and user tries to use same tag for another Revision, an error is raised.
+- Updating a tag for Revision is as simple as providing a new tag for it, as long as the provided tag doesn't exist in traffic block.
+
+
+## Summary
+Following table shows quick summary for traffic splitting flags, their value formats, operation to perform and *Repetition* column denotes if repeating the particular value of flag is allowed in a `kn service update` command.
+
+
+Flag | Value(s) | Operation | Repetition
+:--- | :---: | :--- | :---:
+`--traffic` | `RevisionName=Percent` | Give `Percent` traffic to `RevisionName` | :heavy_check_mark:
+`--traffic` | `Tag=Percent` | Give `Percent` traffic to the Revision having `Tag` | :heavy_check_mark:
+`--traffic` | `@latest=Percent` | Give `Percent` traffic to the latest ready Revision | :heavy_multiplication_x:
+`--tag` | `RevisionName=Tag` | Give `Tag` to `RevisionName` | :heavy_check_mark:
+`--tag` | `@latest=Tag` | Give `Tag` to the latest ready Revision | :heavy_multiplication_x:
+`--untag` | `Tag` | Remove `Tag` from Revision | :heavy_check_mark:
+
+
+## Examples
+1. Tag revisions `echo-v1` and `echo-v2` as `stable` and `staging` respectively:
+```
+kn service update svc --tag echo-v1=stable --tag echo-v2=staging
+```
+
+2. Ramp up/down revision `echo-v3` to `20%`, adjusting other traffic to accommodate:
+```
+kn service update svc --traffic echo-v3=20 --traffic echo-v2=80
+```
+
+3. Give revision `echo-v3` the tag `candidate`, without otherwise changing any traffic split:
+```
+kn service update svc --tag echo-v3=candidate
+```
+
+4. Give `echo-v3` the tag `candidate`, and `2%` of traffic adjusting other traffic to go to revision `echo-v2`:
+```
+kn service update svc --tag echo-v3=candidate --traffic candidate=2 --traffic echo-v2=98
+```
+
+6. Update the tag for `echo-v3` from `candidate` to `current`:
+```
+kn service update svc --untag echo-v3 --tag echo-v3=current
```

Comment:
this should be 

* kn service update svc --untag current --tag echo-v3=current